static/js/mediastream/api.js

```diff
@@ -120,9 +144,25 @@ define(['jquery', 'underscore', 'ua-parser'], function($, _, uaparser) {
 	}
 
 	// Helper hack function to send API requests to other destinations.
-	// Simply provide an alternative send function on the obj Object.
+	// Simply provide an alternative send function on the obj Object. The
+	// alternative send function will get the type and data to send, together
+	// with the original type and data (which are potentially unencrypted).
 	Api.prototype.apply = function(name, obj) {
 		var f = this[name];
+		if (!obj.hasOwnProperty("sendEncrypted")) {
+			obj.sendEncrypted = _.bind(function(type, data) {
+				var to = data.To;
+				if (!to || !this.encryption) {
+					return obj.send(type, data, type, data);
+				}
+
+				this.encryption.encrypt(to, type, data, _.bind(function(encryptedType, encrypted) {
+					encrypted.To = to
+					encrypted.Type = encryptedType
+					obj.send(encryptedType, encrypted, type, data);
+				}, this));
+			}, this);
+		}
```

Comment:
Yep, but the whole `Api.prototype.apply` code/usage in itself is strange ;-)

I'll probably find a way to make that cleaner in a separate PR.